---
layout: default
title: Papers
---

Almost all my papers are indexed at the wonderful
[DBLP](http://www.informatik.uni-trier.de/~ley/db/). Also I have made
available a preprint of my recent papers at
[ArXiv](http://www.arxiv.org/), and I plan to do so for the foreseeable future.

Some publishers also want me to state the obvious: This material is
presented to ensure timely dissemination of scholarly and technical
work. Copyright and all rights therein are retained by authors or by
other copyright holders. All persons copying this information are
expected to adhere to the terms and constraints invoked by each author's
copyright. In most cases, these works may not be reposted without the
explicit permission of the copyright holder.


<h2>Journal Articles</h2>
<table>

<tr valign="top">
<td align="right">
[<a name="Anonymizing-Binary-and-Small-Tables-is-Hard-to-Approximate-journal">1</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, and Riccardo Dondi.
 Anonymizing binary and small tables is hard to approximate.
 <em>Journal of Combinatorial Optimization</em>, 22:97-119, 2011.
[&nbsp;<a href="http://dx.doi.org/10.1007/s10878-009-9277-y">DOI</a>&nbsp;| 
<a href="http://arxiv.org/abs/arXiv:0707.0421">arXiv</a>&nbsp;]
<blockquote>
The problem of publishing personal data without giving up privacy is becoming increasingly important. An interesting formalization recently proposed is the k-anonymity. This approach requires that the rows in a table are clustered in sets of size at least k and that all the rows in a cluster become the same tuple, after the suppression of some records. The natural optimization problem, where the goal is to minimize the number of suppressed entries, is known to be NP-hard when the values are over a ternary alphabet, k=3 and the rows length is unbounded. In this paper we give a lower bound on the approximation factor that any polynomial-time algorithm can achieve on two restrictions of the problem, namely (i) when the records values are over a binary alphabet and k=3, and (ii) when the records have length at most 8 and k=4, showing that these restrictions of the problem are APX-hard.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="bonizzoni10:_approx_clust_finger_vector_missin_values">2</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, and Riccardo Dondi.
 Approximating clustering of fingerprint vectors with missing values.
 <em>Algorithmica</em>, 58(2):282-303, 2010.
[&nbsp;<a href="http://dx.doi.org/10.1007/s00453-008-9265-0">DOI</a>&nbsp;| 
<a href="http://arxiv.org/abs/arXiv:cs/0511082">arXiv</a>&nbsp;]
<blockquote>
The problem of clustering fingerprint vectors with missing values is an interesting problem in Computational Biology that has been proposed in Figueroa et al. (J. Comput. Biol. 11(5):887–901, 2004). In this paper we show some improvements in closing the gaps between the known lower bounds and upper bounds on the approximability of variants of the biological problem. Moreover, we have studied two additional variants of the original problem proposed in Figueroa et al. (Proc. 11th Computing: The Australasian Theory Symposium (CATS), CRPIT, vol. 41, pp. 57–60, 2005). We prove that all such problems are APX-hard even when each fingerprint contains only two unknown positions and we present a greedy algorithm that has constant approximation factors for these variants. Despite the hardness of these restricted versions of the problem, we show that the general clustering problem on an unbounded number of missing values such that they occur for every fixed position of an input vector in at most one fingerprint is polynomial time solvable.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="Beyond-evolutionary-trees">3</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, Tao Jiang, Giulio
  Pavesi, Yuri Pirola, and Lusheng Wang.
 Beyond evolutionary trees.
 <em>Natural Computing</em>, 9(2):421-435, 2010.
[&nbsp;<a href="http://dx.doi.org/10.1007/s11047-009-9156-6">DOI</a>&nbsp;]
<blockquote>
In Computational Biology, the notion of phylogeny has become synonymous with tree-like evolution. Recent advances in the Life Sciences have suggested that evolution has a much more diverse course. In this paper we will survey some of the models that have been proposed to overcome the limitations of using
phylogenies to represent evolutionary histories.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="journal:Pure-Parsimony-Xor-Haplotyping">4</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, Yuri Pirola, and Romeo
  Rizzi.
 Pure parsimony Xor haplotyping.
 <em>IEEE/ACM Transactions on Computational Biology and
  Bioinformatics</em>, 7(4):598-610, 2010.
[&nbsp;<a href="http://doi.ieeecomputersociety.org/10.1109/TCBB.2010.52">DOI</a>&nbsp;| 
<a href="http://arxiv.org/abs/arXiv:1001.1210">arXiv</a>&nbsp;]
<blockquote>
The haplotype resolution from xor-genotype data has been recently formulated as a new model for genetic studies. The xor-genotype data is a cheaply obtainable type of data distinguishing heterozygous from homozygous sites without identifying the homozygous alleles. In this paper we propose a formulation based on a well known model used in haplotype inference: pure parsimony. We exhibit exact solutions of the problem by providing polynomial-time algorithms for some restricted cases and a fixed-parameter algorithm for the general case. These results are based on some interesting combinatorial properties of a graph representation of the solutions. Differently from the same problem on full genotypes, we show that the problem has a polynomial-time k-approximation. Moreover we propose a heuristic and produce an experimental analysis showing that it scales to real-world instances taken from the HapMap project.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="journal:VariantsofConstrainedLongestCommonSubsequence">5</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, and Yuri Pirola.
 Variants of constrained longest common subsequence.
 <em>Information Processing Letters</em>, 110(20):877-881, 2010.
[&nbsp;<a href="http://dx.doi.org/10.1016/j.ipl.2010.07.015">DOI</a>&nbsp;| 
<a href="http://arxiv.org/abs/arXiv:0912.0368">arXiv</a>&nbsp;]
<blockquote>
We consider a variant of the classical Longest Common Subsequence problem called Doubly-Constrained Longest Common Subsequence (DC-LCS).  Given two strings <em>s</em><sub>1</sub> and <em>s</em><sub>2</sub> over an alphabet &Sigma;, a set <em>C</em><sub>s</sub> of strings, and a function <em>C</em><sub>o</sub> : &Sigma;-&gt;<em>N</em>, the DC-LCS problem consists of finding the longest subsequence <em>s</em> of <em>s</em><sub>1</sub> and <em>s</em><sub>2</sub> such that <em>s</em> is a supersequence of all the strings in <em>C</em><sub>s</sub> and such that the number of occurrences in <em>s</em> of each symbol &sigma;in&Sigma; is upper bounded by <em>C</em><sub>o</sub>(&sigma;).  The DC-LCS problem provides a clear mathematical formulation of a sequence comparison problem in Computational Biology and generalizes two other constrained variants of the LCS problem that have been introduced previously in the literature: the Constrained LCS and the Repetition-Free LCS.  We present two results for the DC-LCS problem.  First, we illustrate a fixed-parameter algorithm where the parameter is the length of the solution which is also applicable to the more specialized problems.  Second, we prove a parameterized hardness result for the Constrained LCS problem when the parameter is the number of the constraint strings (|<em>C</em><sub>s</sub>|) and the size of the alphabet &Sigma;.  This hardness result also implies the parameterized hardness of the DC-LCS problem (with the same parameters) and its NP-hardness when the size of the alphabet is constant.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="journal:On-the-Approximation-of-Correlation-Clustering-and-Consensus-Clustering">6</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, and Tao Jiang.
 On the approximation of correlation clustering and consensus
  clustering.
 <em>Journal of Computer and System Sciences</em>, 74(5):671-696, 2008.
[&nbsp;<a href="http://dx.doi.org/10.1016/j.jcss.2007.06.024">DOI</a>&nbsp;| 
<a href="http://arxiv.org/abs/">arXiv</a>&nbsp;]
<blockquote>
The Correlation Clustering problem has been introduced recently as a model for clustering data when a binary relationship between data points is known. More precisely, for each pair of points we have two scores measuring the similarity and dissimilarity respectively, of the two points, and we would like to compute an optimal partition where the value of a partition is obtained by summing up the similarity scores of pairs involving points from the same cluster and the dissimilarity scores of pairs involving points from different clusters. A closely related problem is Consensus Clustering, where we are given a set of partitions and we would like to obtain a partition that best summarizes the input partitions. The latter problem is a restricted case of Correlation Clustering.  In this paper we prove that Minimum Consensus Clustering is APX-hard even for three input partitions, answering an open question in the literature, while Maximum Consensus Clustering admits a PTAS.  We exhibit a combinatorial and practical (4)/(5)-approximation algorithm based on a greedy technique for Maximum Consensus Clustering on three partitions.  Moreover, we prove that a PTAS exists for Maximum Correlation Clustering when the maximum ratio between two scores is at most a constant.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="journal:Exemplar-Longest-Common-Subsequence">7</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, Guillaume Fertin,
  Raffaella Rizzi, and St&eacute;phane Vialette.
 Exemplar longest common subsequence.
 <em>IEEE/ACM Transactions on Computational Biology and
  Bioinformatics</em>, 4(4):535-543, 2007.
[&nbsp;<a href="http://doi.acm.org/10.1145/1322075.1322078">DOI</a>&nbsp;| 
<a href="http://arxiv.org/abs/">arXiv</a>&nbsp;]
<blockquote>
In the paper we investigate the computational and approximation complexity of the Exemplar Longest Common Subsequence of a set of sequences (ELCS problem), a generalization of the Longest Common Subsequence problem, where the input sequences are over the union of two disjoint sets of symbols, a set of mandatory symbols and a set of optional symbols. We show that different versions of the problem are APX-hard even for instances with two sequences.  Moreover, we   show that the related problem of determining the existence of a feasible solution  of the Exemplar Longest Common Subsequence of two sequences is NP-hard.  % On the positive side, efficient algorithms for the ELCS problem over instances of two sequences where each mandatory symbol can appear totally at most three times or the number of mandatory symbols is bounded by a constant are given.  
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="IWBRA2005full">8</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, and Lorenzo Mariani.
 Experimental analysis of a new algorithm for partial haplotype
  completion.
 <em>International Journal of Bioinformatics Research and
  Applications</em>, 1(4):461-473, 2006.
<blockquote>
This paper deals with
the computational problem of inferring complete information on
haplotypes from haplotypes with missing data. This problem is one
of the main issues in <em>haplotyping</em>, as the current DNA
sequencing technology often produces haplotypes with missing bases
and thus the complete information on haplotypes has to be inferred
through computational methods. In this paper we propose a new
algorithmic approach to the problem
 that assumes both the Coalescent and the Minimum
Entropy models and we provide an experimental analysis relating it
to the previously investigated approaches. In particular, the
reconstruction of a perfect phylogeny from haplotypes with missing
data is addressed.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="ReconcilingGeneAndSpeciesTrees">9</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, and Riccardo Dondi.
 Reconciling gene trees to a species tree.
 <em>Theoretical Computer Science</em>, 347(1-2):36-53, 2005.
[&nbsp;<a href="http://gianluca.dellavedova.org/files/papers/ReconcilingGeneAndSpeciesTrees.pdf">.pdf</a>&nbsp;]
<blockquote>
The general problem of reconciling the information from evolutionary trees representing the relationships between distinct gene families is of great importance in Bioinformatics and has been popularized among the Computer Science researchers in (Ma, Li and Zhang, SIAM J. Comp. 2000) where the authors pose the intriguing question if a certain definition of minimum tree that reconciles a gene tree and a species tree is correct. We answer affirmatively to this question; moreover we show an efficient algorithm for computing such  minimum-leaf reconciliation trees and prove the uniqueness of such trees.  We then tackle some different versions of the biological problem by showing that the exemplar problem, arising from the exemplar analysis of multigene genomes, is NP-hard even when the number of copies of a given label is at most two.  Finally we introduce two novel formulations for the problem of recombining evolutionary trees, extending the gene duplication problem studied previously, and we give an exact algorithm (via dynamic programming) for one of these formulations.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="journal:International-Journal-of-Theoretical-Physics">10</a>]
</td>
<td>
Gianpiero Cattaneo, Gianluca Della&nbsp;Vedova, Alberto&nbsp;Ottavio Leporati, and
  Roberto Leporini.
 Towards a theory of conservative computing.
 <em>International Journal of Theoretical Physics</em>, 44(7):861-873,
  2005.
[&nbsp;<a href="http://dx.doi.org/10.1007/s10773-005-7064-9">DOI</a>&nbsp;]
<blockquote>
We extend the notion of conservativeness, given by Fredkin and Toffoli in 1982, to
generic gates whose input and output lines may assume a finite number d of truth
values. A physical interpretation of conservativeness in terms of conservation of the
energy associated to the data used during the computation is given. Moreover, we
define conservative computations, and we show that they naturally induce a new NP-
complete decision problem. Finally, we present a framework that can be used to explicit
the movement of energy occurring during a computation, and we provide a quantum
implementation of the primitives of such framework using creation and annihilation
operators on the Hilbert space <em>C</em><sub>d</sub> , where <em>d</em> is the number of energy levels considered
in the framework.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="INFORMS:MSAasFacilityLocationProblem">11</a>]
</td>
<td>
Winfried Just and Gianluca Della&nbsp;Vedova.
 Multiple sequence alignment as a facility location problem.
 <em>INFORMS Journal on Computing</em>, 16(4):430-440, 2004.
[&nbsp;<a href="http://dx.doi.org/10.1287/ijoc.1040.0093">DOI</a>&nbsp;| 
<a href="http://gianluca.dellavedova.org/files/papers/MSAasFacilityLocationProblem.pdf">.pdf</a>&nbsp;]
<blockquote>
This paper deals with the computational problem of inferring complete information on haplotypes from haplotypes with missing data. This problem is one of the main issues in <em>haplotyping</em>, as the current DNA sequencing technology often produces haplotypes with missing bases and thus the complete information on haplotypes has to be inferred through computational methods. In this paper we propose a new algorithmic approach to the problem that assumes both the Coalescent and the Minimum Entropy models and we provide an experimental analysis relating it to the previously investigated approaches. In particular, the reconstruction of a perfect phylogeny from haplotypes with missing data is addressed.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="journal:2003:A-Library-of-Efficient-Bioinformatics-Algorithms">12</a>]
</td>
<td>
Gianluca Della&nbsp;Vedova and Riccardo Dondi.
 A library of efficient bioinformatics algorithms.
 <em>Applied Bioinformatics</em>, 2(2):117-121, 2003.
[&nbsp;<a href="http://gianluca.dellavedova.org/files/papers/ALiBio-applied-bioinformatics.pdf">.pdf</a>&nbsp;]

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="journal:2003:The-Haplotyping-Problem:-An-Overview-of-Computational-Models-and-Solutions">13</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, and Jing Li.
 The haplotyping problem: An overview of computational models and
  solutions.
 <em>Journal of Computer Science and Technology</em>, 18(6):675-688,
  2003.
[&nbsp;<a href="http://dx.doi.org/10.1007/BF02945456">DOI</a>&nbsp;| 
<a href="http://gianluca.dellavedova.org/files/papers/ALiBio-applied-bioinformatics.pdf">.pdf</a>&nbsp;]
<blockquote>
The investigation of genetic differences among humans has given
evidence that mutations in DNA sequences are responsible for some
genetic diseases. The most common mutation is the one that
involves only a single nucleotide of the DNA sequence, which is
called a <em>single nucleotide polymorphism (SNP)</em>. As a
consequence, computing a complete map of all SNPs occurring in the
human populations is one of the primary goals of recent studies in
human genomics.
The construction of such a map requires to determine the DNA
sequences that from all chromosomes. In diploid organisms like
humans, each chromosome consists of two sequences called
<em>haplotypes</em>. Distinguishing the information contained in
both haplotypes when analyzing  chromosome sequences poses several
new computational issues which collectively form a new emerging
topic of Computational Biology known as <em>Haplotyping</em>.
This paper is a comprehensive study of some  new combinatorial
approaches proposed in this research area and it mainly  focuses
on the formulations and algorithmic solutions of some  basic
biological problems. Three statistical approaches are briefly
discussed at the end of the paper.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="LocalVertex">14</a>]
</td>
<td>
Gianluca Della&nbsp;Vedova and H.&nbsp;Todd Wareham.
 Optimal algorithms for local vertex quartet cleaning.
 <em>Bioinformatics</em>, 18(10):1297-1304, 2002.
<blockquote>
Reconstructing evolutionary trees is an important problem in biology. A response to the computational intractability of most of the traditional criteria for inferring evolutionary trees has been a focus on new criteria, particularly quartet-based methods that seek to merge trees derived on subsets of four species from a given species-set into a tree for that entire set.  Unfortunately, most of these methods are very sensitive to errors in the reconstruction of the trees for individual quartets of species. A recently-developed technique called <em>quartet cleaning</em> can alleviate this difficulty in certain cases by using  redundant information in the complete set of quartet topologies for a given species-set to correct such errors. In this paper, we describe two new local vertex quartet cleaning algorithms which have optimal time complexity and error-correction bound, respectively. These are the first known local vertex quartet cleaning algorithms that are optimal with respect to either of these attributes.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="valinsky02:_analysis">15</a>]
</td>
<td>
Lea Valinsky, Gianluca Della&nbsp;Vedova, A.J. Scupham, S.&nbsp;Alvey, Andres Figueroa,
  B.&nbsp;Yin, R.J. Hartin, Marek Chrobak, D.E. Crowley, Tao Jiang, and James
  Borneman.
 Analysis of bacterial microbial community composition by
  oligonucleotide fingerprinting of rrna genes.
 <em>Applied and Environmental Microbiology</em>, 68(7):3243-3250, 2002.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="valinsky02:_oligon_finger_genes_analy_fungal_commun_compos">16</a>]
</td>
<td>
Lea Valinsky, Gianluca Della&nbsp;Vedova, Tao Jiang, and James Borneman.
 Oligonucleotide fingerprinting of rrna genes for analysis of fungal
  community composition.
 <em>Applied and Environmental Microbiology</em>, 68(12):5999-6004,
  2002.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="experimentalLCS">17</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, and Giancarlo Mauri.
 Experimenting an approximation algorithm for the LCS.
 <em>Discrete Applied Mathematics</em>, 110(1):13-24, 2001.
[&nbsp;<a href="http://dx.doi.org/10.1016/S0166-218X(00)00300-0">DOI</a>&nbsp;]
<blockquote>
The problem of finding the longest common subsequence (lcs) of a given set of sequences over an alphabet &Sigma; occurs in many interesting contexts, such as data compression and molecular biology, in order to measure the &ldquo;similarity degree&rdquo;   among biological sequences. Since the problem is NP-complete in its decision version  (i.e. does there exist a lcs of length at least <em>k</em>, for a given <em>k</em>?) even over fixed alphabet, polynomial algorithms which give approximate solutions have been proposed. Among them, Long Run (LR) is the only one with guaranteed constant performance ratio.
In this paper, we give a new approximation algorithm for the longest common subsequence problem: the Expansion algorithm (EA). First of all we prove that the solution found by the Expansion algorithm is always at least as good as the one found by LR. Then we report the results of an experimentation with two different groups of instances, which show that EA clearly outperforms Long Run in practice.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="ProbeSelectionForMicrobialCommunities">18</a>]
</td>
<td>
James Borneman, Marek Chrobak, Gianluca Della&nbsp;Vedova, Andres Figueroa, and Tao
  Jiang.
 Probe selection algorithms with applications in the analysis of
  microbial communities.
 <em>Bioinformatics</em>, 17(Suppl. 1):S39-S48, 2001.
[&nbsp;<a href="http://gianluca.dellavedova.org/files/papers/ProbeSelectionForMicrobialCommunities.pdf">.pdf</a>&nbsp;]
<blockquote>
We propose two efficient heuristics for minimizing the number of oligonucleotide probes needed for analyzing populations of ribosomal RNA gene (rDNA) clones by hybridization experiments on DNA microarrays.  Such analyses have applications in the study of microbial communities. Unlike in the classical SBH (sequencing by hybridization) procedure, where multiple probes are on a DNA chip, in our applications we perform a series of experiments, each one consisting of applying a single probe to a DNA microarray containing a large sample of rDNA sequences from the studied population.  The overall cost of the analysis is thus roughly proportional to the number of experiments, underscoring the need for minimizing the number of probes. Our algorithms are based on two well-known optimization techniques, <i>i.e.</i> simulated annealing and Lagrangian relaxation, and our preliminary tests demonstrate that both algorithms are able to find satisfactory probe sets for real rDNA data.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="MSASPNPcomplete">19</a>]
</td>
<td>
Paola Bonizzoni and Gianluca Della&nbsp;Vedova.
 The complexity of multiple sequence alignment with SP-score that is
  a metric.
 <em>Theoretical Computer Science</em>, 259(1):63-79, 2001.
[&nbsp;<a href="http://gianluca.dellavedova.org/files/papers/MSASPNP-complete.pdf">.pdf</a>&nbsp;]
<blockquote>
This paper analyzes the computational complexity of computing the optimal alignment of a set of sequences under the SP (sum of all pairs) score scheme.  We  solve an open question  by showing that  the problem is <em>NP</em>-complete in the very restricted case in which the sequences are over a binary alphabet  and the score is a metric.  This result establishes the intractability of multiple sequence alignment under a score function of  mathematical interest, which has indeed received much attention in biological sequence comparison.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="inapproximableMITjournal">20</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, and Giancarlo Mauri.
 Approximating the maximum isomorphic agreement subtree is hard.
 <em>International Journal on the Foundations of Computer Science</em>,
  11(4):579-590, 2000.
[&nbsp;<a href="http://gianluca.dellavedova.org/files/papers/inapproximableMITjournal.pdf">.pdf</a>&nbsp;]
<blockquote>
The Maximum Isomorphic Agreement Subtree (MIT) problem is one of the simplest versions of the Maximum Interval Weight Agreement Subtree method (MIWT) which is used to compare phylogenies. More precisely MIT allows to provide a subset of the species such that the exact distances between species in such subset are preserved among all evolutionary trees considered.  In this paper, the approximation complexity of the MIT problem is investigated, showing that it  cannot be approximated in polynomial time within factor  log<sup>&delta;</sup><em>n</em> for any &delta;&gt;0  unless NP=DTIME(2<sup>polylog n</sup>) for instances containing three trees. Moreover, we show that such result can be strengthened whenever instances of the MIT problem can contain an arbitrary number of  trees, since MIT shares the same approximation lower bound of MAX CLIQUE.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="ModularDecompositionHypergraphs">21</a>]
</td>
<td>
Paola Bonizzoni and Gianluca Della&nbsp;Vedova.
 An algorithm to compute the modular decomposition of hypergraphs.
 <em>Journal of Algorithms</em>, 32(2):65-86, 1999.
[&nbsp;<a href="http://gianluca.dellavedova.org/files/papers/ModularDecompositionHypergraphs.pdf">.pdf</a>&nbsp;]
<blockquote>
  We propose an <em>O</em>(<em>n</em><sup>4</sup>) algorithm to build the modular decomposition tree of hypergraphs of dimension 3 and show how this algorithm can be generalized to compute in <em>O</em>(<em>n</em><sup>3k -5</sup>) time the decomposition of hypergraphs of any fixed dimension <em>k</em>.
</blockquote>

</td>
</tr>
</table>
<h2>Books and Books Chapters </h2>
<table>

<tr valign="top">
<td align="right">
[<a name="book:2008:The-Comparison-of-Phylogenetic-Networks:-Algorithms-and-Complexity">1</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, and Giancarlo Mauri.
 The comparison of phylogenetic networks: Algorithms and complexity.
 In Ion Mandoiu and Alexander Zelikovsky, editors, <em>Bioinformatics
  Algorithms: Techniques and Applications</em>, pages 143-173. Wiley-Interscience,
  2008.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="GSTA07">2</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, and Jing Li.
 The haplotyping problem: An overview of computational models and
  solutions.
 In Sun Kim, Haixu Tang, and Elaine&nbsp;R. Mardis, editors, <em>Genome
  Sequencing Technology and Algorithms</em>, pages 151-181. Artech House, Inc.,
  2007.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="MolecularMicrobialEcologyManual">3</a>]
</td>
<td>
Lea Valinsky, Alexandra&nbsp;J. Scupham, Gianluca Della&nbsp;Vedova, Zheng Liu, Andres
  Figueroa, Katechan Jampachaisri, Bei Yin, Elizabeth Bent, Robert
  Mancini-Jones, James Press, Tao Jiang, and James Borneman.
 Oligonucleotide fingerprinting of ribosomal rna genes (ofrg).
 In G.&nbsp;A. Kowalchuk, F.&nbsp;J. de&nbsp;Bruijn, I.&nbsp;M. Head, A.&nbsp;D.&nbsp;L. Akkermans,
  and J.&nbsp;D. van Elsas, editors, <em>Molecular Microbial Ecology Manual</em>, pages
  569-585. Kluwer Academic Publishers, Dordrecht, The Netherlands, 2nd
  edition, 2004.

</td>
</tr>
</table>
<h2>Papers in Conference Proceedings</h2>
<table>

<tr valign="top">
<td align="right">
[<a name="conference:1995:Parameterized-Complexity-of-k-Anonymity:-Hardness-and-Tractability">1</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, and Yuri Pirola.
 Parameterized complexity of <em>k</em>-anonymity: Hardness and tractability.
 In <em>Proc. 21st International Workshop on Combinatorial Algorithms
  (IWOCA 2010)</em>, volume 6460 of <em>LNCS</em>, pages 242-255, 2011.
[&nbsp;<a href="http://dx.doi.org/10.1007/978-3-642-19222-7_25">DOI</a>&nbsp;]

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="conference:PIntronICCABS">2</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Yuri Pirola, and Raffaella Rizzi.
 Pintron: A fast method for gene structure prediction via maximal
  pairings of a pattern and a text.
 In <em>Computational Advances in Bio and Medical Sciences (ICCABS),
  2011 IEEE 1st International Conference on</em>, pages 33 -39, feb. 2011.
[&nbsp;<a href="http://dx.doi.org/10.1109/ICCABS.2011.5729935">DOI</a>&nbsp;| 
<a href="http://arxiv.org/abs/arxiv:1005.1514">arXiv</a>&nbsp;]
<blockquote>
A challenging issue in designing computational methods for predicting the gene structure into exons and introns from a cluster of transcript (EST, mRNA) sequences, is guaranteeing both accuracy and efficiency in time and space, when large clusters of over than 20,000 ESTs and genes longer than 1Mb are processed. Traditionally, the problem has been faced by combining different tools, not specifically designed for this task. We propose a fast method based on ad hoc procedures for solving the problem. Our method combines two ideas: a novel algorithm of proved small time complexity for computing spliced alignments of a transcript against a genome, and an efficient algorithm that exploits the inherent redundancy of information in a cluster of transcripts to select, among all possible factorizations of EST sequences, those allowing to infer splice site junctions that are largely confirmed by the input data. The EST alignment procedure is based on the construction of maximal embeddings, that are sequences obtained from paths of a graph structure, called embedding graph, whose vertices are the maximal pairings of a genomic sequence T and an EST P. The procedure runs in time linear in the length of P and T and in the size of the output. PIntron, the software tool implementing our methodology, is available at http://www.algolab.eu/PIntron and it is able to process in a few seconds some critical genes that are not manageable by other gene structure prediction tools. At the same time, PIntron exhibits high accuracy (sensitivity and specificity) when compared with ENCODE data.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="conference:Minimum-Factorization-Agreement-of-Spliced-ESTs">3</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, Yuri Pirola, and
  Raffaella Rizzi.
 Minimum factorization agreement of spliced ests.
 In <em>Proceedings of the 9th Workshop on Algorithms in
  Bioinformatics (WABI 2009)</em>, pages 1-12, 2009.
[&nbsp;<a href="http://arxiv.org/abs/">arXiv</a>&nbsp;]
<blockquote>
Producing <em>spliced EST</em> sequences is a fundamental task in
the computational problem of reconstructing    splice  and
transcript variants,  a crucial step in the alternative splicing
investigation.  Now, given an EST sequence, there can be
several spliced EST sequences associated to it, since
the original EST sequences may have different  alignments against wide genomic regions.
In this paper we address a crucial issue arising from  the above
step: given a collection <em>C</em>  of different spliced EST
sequences that are associated to  an initial set <em>S</em> of EST
sequences, how can we extract a subset <em>C</em>'  of <em>C</em> such that
each EST sequence in <em>S</em> has a putative spliced EST in <em>C</em>' and
<em>C</em>' agree on a common alignment region  to the genome or gene
structure?
We introduce a new computational problem that models the above issue, and at
the same time is also relevant in some more
general settings, called <em>Minimum Factorization
Agreement</em> (MFA).
We investigate some algorithmic solutions  of the MFA problem and
their  applicability  to real data sets. We   show that algorithms solving the MFA
problem are able to find efficiently the correct spliced EST
associated to an EST even when the splicing of sequences is obtained
by a rough alignment process. Then we show that   the MFA
method could be used in producing or analyzing spliced EST
libraries under various biological criteria.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="conference:A-PTAS-for-the-Minimum-Consensus-Clustering-Problem-with-a-Fixed-Number-of-Clusters">4</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, and Riccardo Dondi.
 A PTAS for the minimum consensus clustering problem with a fixed
  number of clusters.
 In <em>Proceedings of the 11th Italian Conference on Theoretical
  Computer Science (ICTCS 2009)</em>, 2009.
[&nbsp;<a href="http://arxiv.org/abs/arXiv:0907.1840">arXiv</a>&nbsp;| 
<a href="http://gianluca.dellavedova.org/files/papers/ICTCS2009-talk.pdf">Slides</a>&nbsp;]
<blockquote>
The Consensus Clustering problem has been introduced as an effective
way to analyze the results of different  microarray experiments.
The problem  consists of looking for a partition that best
summarizes  a set of input partitions
under a simple cost
function.
We investigate the restriction of Consensus Clustering
when the output partition is required to contain at most <em>k</em> sets, giving a
polynomial time approximation scheme (PTAS).
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="conference:2010:The-k-anonymity-Problem-is-Hard">5</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, and Yuri Pirola.
 The <em>k</em>-anonymity problem is hard.
 In <em>Proceedings of the 17th International Symposium on
  Fundamentals of Computation Theory (FCT 2009)</em>, pages 26-37, 2009.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="conference:Pure-Parsimony-Xor-Haplotyping">6</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, Yuri Pirola, and Romeo
  Rizzi.
 Pure parsimony xor haplotyping.
 In <em>Proceedings of the 5th International Symposium on
  Bioinformatics Research and Applications, (ISBRA 2009)</em>, volume 5542 of <em>
  LNCS</em>, pages 186-197, 2009.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="conference:2006:Fingerprint-Clustering-with-Bounded-Number-of-Missing-Values">7</a>]
</td>
<td>
Paola Bonizzoni, Gianluca&nbsp;Della Vedova, Riccardo Dondi, and Giancarlo Mauri.
 Fingerprint clustering with bounded number of missing values.
 In <em>Proceedings of the 17th Symposium on Combinatorial Pattern
  Matching (CPM2006)</em>, pages 106-116, 2006.
[&nbsp;<a href="http://dx.doi.org/10.1007/11780441_11">DOI</a>&nbsp;| 
<a href="http://arxiv.org/abs/arXiv:cs/0511082">arXiv</a>&nbsp;]

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="IWBRA06">8</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, Guillaume Fertin, and
  St&eacute;phane Vialette.
 Exemplar longest common subsequence.
 In <em>International Conference on Computational Science (2)</em>, pages
  622-629, 2006.
[&nbsp;<a href="http://dx.doi.org/10.1007/11758525_85">DOI</a>&nbsp;]

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="conference:A-PTAS-for-the-Maximum-Consensus-Clustering">9</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, and Giancarlo Mauri.
 A PTAS for the maximum consensus clustering.
 In <em>Proc. 1st FIMA International Conference Models and Methods
  for Human Genomics</em>, 2006.
[&nbsp;<a href="http://arxiv.org/abs/">arXiv</a>&nbsp;]

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="IWBRA2005">10</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, and Lorenzo Mariani.
 Experimental analysis of a new algorithm for partial haplotype
  completion.
 In <em>International Conference on Computational Science (2)</em>,
  volume 3515 of <em>LNCS</em>, pages 952-959, 2005.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="Isaac2005Clustering">11</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Riccardo Dondi, and Tao Jiang.
 Correlation clustering and consensus clustering.
 In <em>Algorithms and Computation, 16th International Symposium
  (ISAAC)</em>, volume 3827 of <em>LNCS</em>, pages 226-235, 2005.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="pozzi04:_explic_upper_bound_approx_ratio">12</a>]
</td>
<td>
Sergio Pozzi, Gianluca Della&nbsp;Vedova, and Giancarlo Mauri.
 An explicit upper bound for the approximation ratio of the maximum
  gene regulatory network problem.
 In <em>CMSB04</em>, volume 3082 of <em>LNCS</em>, pages 1-8, 2004.
<blockquote>
One of the combinatorial models for the biological problem of inferring gene
regulation networks is the  Maximum Gene Regulatory Network Problem,
shortly MGRN. The
problem is <b>NP</b>-hard, consequently the attention has shifted
towards
approximation algorithms, leading to a polynomial-time 1/2-approximation
algorithm, while no upper bound on the
possible approximation ratio
was previously known.
In this paper we make a first step towards closing the gap between the best
known and the best possible approximation factors, by showing that no
polynomial-time approximation algorithm can have a factor better than
1-(1/8)/(1+<em>e</em><sup>2</sup>)
unless <b>RP</b>=<b>NP</b>.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="ciac2003">13</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, and Riccardo Dondi.
 Reconciling gene trees to a species tree.
 In R.&nbsp;Petreschi, G.&nbsp;Persiano, and R.&nbsp;Silvestri, editors, <em>
  Algorithms and Complexity, Proceedings of the 5th Italian Conference (CIAC
  2003)</em>, volume 2653 of <em>LNCS</em>, pages 120-131, 2003.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="sac2002">14</a>]
</td>
<td>
Gianluca Della&nbsp;Vedova and H.&nbsp;Todd Wareham.
 Optimal algorithms for local vertex quartet cleaning.
 In <em>Proceedings of the 17th ACM Symposium on Applied Computing
  (SAC2002)</em>, pages 173-177, 2002.
[&nbsp;<a href="http://gianluca.dellavedova.org/files/papers/LocalVertex.pdf">.pdf</a>&nbsp;]

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="soda2002">15</a>]
</td>
<td>
Gianluca Della&nbsp;Vedova, Tao Jiang, Jing Li, and Jianjun Wen.
 Approximating minimum quartet inconsistency.
 In <em>13th Annual ACM-SIAM Symposium on Discrete Algorithms
  (SODA2002)</em>, pages 894-895, 2002.
[&nbsp;<a href="http://gianluca.dellavedova.org/files/papers/soda2002.pdf">.pdf</a>&nbsp;]

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="ProbeSelectionForMicrobialCommunities-ismb2001">16</a>]
</td>
<td>
James Borneman, Marek Chrobak, Gianluca Della&nbsp;Vedova, Andres Figueroa, and Tao
  Jiang.
 Probe selection algorithms with applications in the analysis of
  microbial communities.
 In <em>Proceedings of the 9th International Conference on
  Intelligent Systems for Molecular Biology (ISMB2001)</em>, pages S39-S48, 2001.
[&nbsp;<a href="http://gianluca.dellavedova.org/files/papers/ismb2001_talk.pdf">Slides</a>&nbsp;]

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="phytopatology1">17</a>]
</td>
<td>
Lea Valinsky, A.J. Scupham, Gianluca Della&nbsp;Vedova, Marek Chrobak, Tao Jiang,
  Andres Figueroa, R.J. Hartin, B.&nbsp;Yin, and James Borneman.
 A DNA array approach for analysis of microbial communities using
  oligonucleotide fingerprinting of ribosomal RNA genes.
 In <em>Annual Meeting of the American Phytopathological Society
  2001</em>, volume&nbsp;91 of <em>Phytopathology</em>, page S90, 2001.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="sac2001">18</a>]
</td>
<td>
Paolo Barone, Paola Bonizzoni, Gianluca Della&nbsp;Vedova, and Giancarlo Mauri.
 An approximation algorithm for the shortest common supersequence: An
  experimental analysis.
 In <em>Proceedings of the 16th ACM Symposium on Applied Computing
  (SAC2001)</em>, pages 56-60, 2001.
[&nbsp;<a href="http://gianluca.dellavedova.org/files/papers/sac2001.pdf">.pdf</a>&nbsp;]
<blockquote>
In this paper an approximation algorithm, called Reduce-Expand,
for the Shortest Common Supersequence (SCS) problem is presented
and its behavior is studied experimentally. While the guaranteed
approximation ratio of Reduce-Expand matches that of the best
known algorithm, our algorithm clearly outperforms such algorithm
with respect to the length of the approximate solution.
</blockquote>

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="MSAasFacilityLocationProblem">19</a>]
</td>
<td>
Winfried Just and Gianluca Della&nbsp;Vedova.
 Multiple sequence alignment as a facility location problem.
 In <em>Proceedings of the Prague Stringology Club Workshop 2000
  (PSCW2000)</em>, 2000.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="cpm2000">20</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, and Giancarlo Mauri.
 Approximating the maximum isomorphic agreement subtree is hard.
 In Raffaele Giancarlo and David Sankoff, editors, <em>Proceedings of
  the 11th Symposium on Combinatorial Pattern Matching (CPM2000)</em>, volume 1848
  of <em>LNCS</em>, pages 119-128, 2000.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="alex98">21</a>]
</td>
<td>
Paola Bonizzoni, Massimo D'Alessandro, Gianluca Della&nbsp;Vedova, and Giancarlo
  Mauri.
 Experimenting an approximation algorithm for the lcs.
 In Roberto Battiti and Alan Bertossi, editors, <em>Algorithms and
  Experiments (ALEX98)</em>, pages 96-102, 1998.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="conference:1995:Modular-Decomposition-of-Hypergraphs">22</a>]
</td>
<td>
Paola Bonizzoni and Gianluca Della&nbsp;Vedova.
 Modular decomposition of hypergraphs.
 In Manfred Nagl, editor, <em>Proceedings of the 21st Workshop on
  Graph-Theoretic Concepts in Computer Science (WG95)</em>, volume 1017 of <em>
  LNCS</em>, pages 303-317, 1995.

</td>
</tr>
</table>
<h2>Other</h2>
<table>

<tr valign="top">
<td align="right">
[<a name="JCST:SpecialIssue2004Foreword">1</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, and Tao Jiang.
 Foreword - special issue on bioinformatics.
 <em>Journal of Computer Science and Technology</em>, 19(1):1-1, 2004.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="trace1">2</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, and Tao Jiang.
 Approximating maximum trace alignment and its complement.
 manuscript, 1999.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="TR201">3</a>]
</td>
<td>
Paola Bonizzoni, Gianluca Della&nbsp;Vedova, Dario Lucarella, and Giancarlo Mauri.
 On genericity and stability in a graph based data model.
 Technical Report TR 201-97, Dipartimento di Scienze
  dell'Informazione, Universit&agrave; degli Studi di Milano, 1997.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="manus">4</a>]
</td>
<td>
P.&nbsp;Bonizzoni, G.&nbsp;Della&nbsp;Vedova, D.&nbsp;Lucarella, and G.&nbsp;Mauri.
 Graal: a graph based query language.
 manuscript, 1997.

</td>
</tr>


<tr valign="top">
<td align="right">
[<a name="TR139">5</a>]
</td>
<td>
Gianluca Della&nbsp;Vedova.
 An algorithm to compute the modular decomposition of k-structures.
 Technical Report TR139-95, Universit&agrave; degli Studi di Milano,
  Dipartimento di Scienze dell'Informazione, 1995.

</td>
</tr>
</table>